### ISSUE TITLE
Change array initialization syntax

### ISSUE THREAD

**Issue by danieldaeschle** [vlang/v#7061]
Currently we have this kind of syntax:
```
_ := [u32(10), 5, 10]
```
The array type is inferred to `[]u32`
The syntax should change to a cast like initialization:
```
_ := []u32([10, 5, 10])
```

This should also work for sum types.

**Comment by Delta456** [comment]
Will the latter one not make it look like a function?

**Comment by danieldaeschle** [comment]
no, like a cast.

**Comment by Delta456** [comment]
Ok, it is casted to `[]u32()` like how it is done in Go for `[]byte` and `[]rune`.

**Comment by spaceface777** [comment]
yep, so rather than casting the first element of the array, this new syntax would cast the array itself.

**Comment by spytheman** [comment]
what would happen for this?
```v
a := [1,2,3]
b := []u64(a)
```

**Comment by danieldaeschle** [comment]
@spytheman i think a is array of `any_int` and b is array of `u64`

**Comment by spytheman** [comment]
a is an array of just ints, my question is more about what should []u64(a) do:
1) produce a checker error
2) generate code for looping over a, and producing a new array of u64 elements.

A cast usually does no looping at all, it just changes how the underlying data is treated, but I do not see how will that work for arrays.

**Comment by danieldaeschle** [comment]
How does this behave?
```v
a := 1
b := u64(a)
```

**Comment by spytheman** [comment]
a cast of a single element just changes how it is handled; it does not change the data itself, and is easy to do
a cast of an entire array has potentially large ramifications

**Comment by danieldaeschle** [comment]
What would be your suggestion?

**Comment by danieldaeschle** [comment]
checker error might be okay. let's see what @medvednikov says.

**Comment by ntrel** [comment]
+1 for checker error, only array literals should be castable.

**Comment by medvednikov** [comment]
It's an interesting problem. Go solves it by not allowing to cast between arrays :)

We can do the same.

**Comment by MarcTCruz** [comment]
I vote for current syntax to kep: 
`_ := [u32(10), 5, 10]`

Less verbose and clear enough.

**Comment by ushakov-s** [comment]
What if we use 
```v
var x [3]u32
```
for empty arrays and
```v
_ := u32[10, 5, 10]
```
for array initialization.
This prevents necessary of any casting and makes code more readable.

**Comment by JalonSolov** [comment]
`x [3]u32` is too close to the syntax for creating a fixed-size array, which would be `x := [3]u32{}`.

It is very different from `[u32(10), 5, 10]`, which is a dynamic array that happens to be initialized with 3 values.

Introducing a change to look like a cast introduces other issues.

I'm not sure why we need a change at all.
